models/files.go

```diff
@@ -1,15 +1,61 @@
 package models
 
 import (
+	"bytes"
 	"fmt"
 	"path/filepath"
+	"runtime"
 	"strings"
 
 	"github.com/natsukagami/kjudge/db"
 	"github.com/natsukagami/kjudge/models/verify"
 	"github.com/pkg/errors"
 )
 
+func crlftoLF(content []byte) ([]byte, error) {
+	return bytes.ReplaceAll(content, []byte("\r\n"), []byte("\n")), nil
+}
+
+func lftoCRLF(content []byte) ([]byte, error) {
+	lf := bytes.Count(content, []byte("\n"))
+	crlf := bytes.Count(content, []byte("\r\n"))
+	if crlf == lf {
+		return content, nil
+	}
+	var err error = nil
+	if crlf != 0 {
+		err = errors.Errorf("number of crlf and lf (%v, %v) does not match", crlf, lf)
+	}
+	return bytes.ReplaceAll(content, []byte("\r\n"), []byte("\n")), err
+}
+
+// NormalizeEndings normalize file line endings to the target OS's endings
+// target accepts "windows" or "linux". Returns error if OS is not supported
+// or there is LF and CRLF mixed together
+func NormalizeEndings(content []byte, target string) ([]byte, error) {
+	switch target {
+	case "windows":
+		return lftoCRLF(content)
+	case "linux":
+		return crlftoLF(content)
+	default:
+		return nil, errors.Errorf("%s not supported for line ending conversion", runtime.GOOS)
+	}
+}
+
+// NormalizeEndings normalize file line endings to the current OS's endings
+// target accepts "windows" or "linux"
+func NormalizeEndingsNative(content []byte) ([]byte, error) {
+	return NormalizeEndings(content, runtime.GOOS)
+}
```

Comment:
We can also do this with [conditional compilation](https://blog.ralch.com/articles/golang-conditional-compilation/).